@vates/generator-toolbox/src/synchronized.test.mts

```diff
@@ -0,0 +1,142 @@
+import assert from 'node:assert'
+import { suite, test } from 'node:test'
+import { Synchronized } from './synchronized.mjs'
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+async function* makeRangeGenerator(end = Infinity, progress = { yielded: 0 }, onYielded = (val: unknown) => {}) {
+  for (let i = 0; i < end; i++) {
+    await new Promise(resolve => setTimeout(resolve, 10))
```

Comment:
I think you should use [mocked timers](https://nodejs.org/api/test.html#timers) in your test

Reply:
I didn't succeed in using it in this case, since artificially advancing the timer also impact block generation